src/Commands/core/UpdateDBCommands.php

```diff
@@ -364,8 +364,11 @@ public static function updateDoOnePostUpdate($function, DrushBatchContext $conte
      */
     public static function updateFinished($success, $results, $operations)
     {
-        // No code needed but the batch result bookkeeping fails without a finished callback.
-        $noop = 1;
+        // Flush all caches at the end of the batch operation. When Drupal core
+        // performs database updates it also clears the cache at the end. This
+        // ensures that we are compatible with updates that rely on this
+        // behavior.
+        drupal_flush_all_caches();
```

Comment:
Should only run if the option --cache-clear is true. PR can be merged after thats done.